scripts/ui/inventory/hotbar.gd

```diff
@@ -0,0 +1,104 @@
+extends UserInterface
+
+const SlotClass = preload("res://scripts/ui/inventory/slot.gd") # Preload the Slot script
+@onready var hotbar = $HotbarSlots # Reference to the HotbarSlots node
+@onready var slots = $HotbarSlots.get_children() #List of all child nodes of HotbarSlots
+@onready var active_item_label = $ActiveItemLabel # Reference to the ActiveItemLabel node
+@onready var tooltip = $ActiveItemLabel/Tooltip # Reference to the Tooltip node
+var t
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	#Sets up the Hotbar
+	PlayerInventory.active_item_updated.connect(self.update_active_item_label)
+	for i in range(slots.size()):
+		slots[i].gui_input.connect(slot_gui_input.bind(slots[i]))
+		PlayerInventory.active_item_updated.connect(slots[i].refresh_style)
+		slots[i].slot_index = i
```

Comment:
Not sure why slots need to know their id, should be managed in the parent handler script?